scipy/stats/_continuous_distns.py

```diff
@@ -4086,6 +4086,89 @@ def _stats(self, a):
         return outputs[:]
 kappa3 = kappa3_gen(a=0.0, name='kappa3')
 
+class moyal_gen(rv_continuous):
+    r"""A Moyal continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `moyal` is::
+
+    .. math::
+
+        f(x) = \exp{-\frac{1}{2}(x+\exp{-x})} / \sqrt(2\pi)
+
+    `moyal` takes ``loc`` as a location parameter and ``scale``
+    as a scale parameter. Here :math:`x` is the usual standardized form.
+
+    This distribution has utility in high-energy physics and radiation
+    detection. It describes the energy loss of a charged relativistic
+    particle due to ionization of the medium [1]_. It also provides an
+    approximation for the Landau distribution. For an in depth description
+    see [2]_. For additional description, see [3]_.
+
+    References
+    ----------
+    .. [1] J.E. Moyal, "XXX. Theory of ionization fluctuations",
+           The London, Edinburgh, and Dublin Philosophical Magazine
+           and Journal of Science, vol 46, 263-280, (1955).
+           https://doi.org/10.1080/14786440308521076 (gated)
+    .. [2] G. Cordeiro et al., "The beta Moyal: a useful skew distribution",
+           International Journal of Research and Reviews in Applied Sciences,
+           vol 10, 171-192, (2012).
+           http://www.arpapress.com/Volumes/Vol10Issue2/IJRRAS_10_2_02.pdf
+    .. [3] C. Walck, "Handbook on Statistical Distributions for
+           Experimentalists; International Report SUF-PFY/96-01", Chapter 26,
+           University of Stockholm: Stockholm, Sweden, (2007).
+           www.stat.rice.edu/~dobelman/textfiles/DistributionsHandbook.pdf
+
+    %(after_notes)s
+
+    .. versionadded:: 1.1.0
+
+    %(example)s
+
+    """
+    def _rvs(self):
+        sz, rndm = self._size, self._random_state
+        u1 = gamma.rvs(a = 0.5, scale = 2, size=sz, random_state=rndm)
+        return -np.log(u1)
+
+    def _pdf(self, x):
+        return np.exp(-0.5 * (x + np.exp(-x))) / np.sqrt(2*np.pi)
+
+    def _cdf(self, x):
+        return sc.erfc(np.exp(-0.5 * x) / np.sqrt(2))
+
+    def _sf(self, x):
+        return sc.erf(np.exp(-0.5 * x) / np.sqrt(2))
+
+    def _ppf(self, x):
+        return -np.log(2 * sc.erfcinv(x)**2)
+
+    def _stats(self):
+        mu = np.log(2) + np.euler_gamma
+        mu2 = np.pi**2 / 2
+        g1 = 28 * np.sqrt(2) * sc.zeta(3) / np.pi**3
+        g2 = 4.
+        return mu, mu2, g1, g2
+
+    def _munp(self, n):
```

Comment:
Hi, I might be wrong but it look like _munp returns the _central_ moments if n=2,3,4 whereas stats.moments should return _non-central_ moments (see 26.4 in http://www.stat.rice.edu/~dobelman/textfiles/DistributionsHandbook.pdf).

Christoph

Reply:
Hi Christoph,

Thanks for your comments.

Hmm, oops, perhaps I made a mistake in `_munp`. For these I'm assuming the scaling/shifting is handled automatically based on the central moments. Otherwise it would be necessary to explicitly multiply the `_munp` parameters by appropriate powers of the scale parameter, etc. Should `_stats` really use raw moments? Perhaps an expert can weigh in here.

Reply:
stats is fine and should return mvsk, mean, variance (centered second moment) skew and kurtosis

munp (if that is the right function name) should return non-centered moments. Note centering on mean is independent of loc scale in asymmetric distributions like this, super class takes care of loc and scale.
But in asymmetric distribution, mean is not zero, even if loc =0.

What is strange is that a bug here should have been caught by the generic unit tests. I haven't looked much at the details in recent time, but this is a part that can be verified generically.

Reply:
Hi,

yes, it is not about `loc` or `scale`. munp should return E[X^k] and not E[(X - mean(X))^k] (central moment). In the handbook referenced above, the former is called mu_k' and the latter simply mu_k (section 2.2) and section 26.4 on the moyal distribution gives an expression for mu_2 and mu_3 (i.e. the centered one). (You can derive the non-centered moment by expanding the square / cube)

Reply:
Ok, I have attempted to put in the correct analytic raw moments. With these changes if I look at a test distribution I get the following:

```
In [1]: import scipy.stats as ss
In [2]: mo=ss.moyal(loc=2,scale=2)
In [3]: for i in range(2,5):
    print "%.16f" %mo.moment(i)
    print "%.16f" %mo.expect(lambda x: x**i)
    print "  "
   ....:     
40.3573986023864748
40.3573986023866880
  
497.1429148596399727
497.1429148596396317
  
8039.7631977583432672
8039.7631977583487242
```

Reply:
OK, Looks good. I don't know if the rather long expressions in case `n==3` or `n == 4` add much value since the expectation can be computed also numerically and the results are the same up to ~9 digits. `n==3`: note that you sometimes did not put spaces around the operator `+`. 
But apart from such very minor things, I think the implementation looks good.

Reply:
In an earlier comment Josef mentioned first few raw moments should be available in closed form. I guess my definition of "few" was first four moments.

The expressions are long, but the speed-up should be significant compared to integration. I will fix the formatting around the operators, good catch.